Il2CppInterop.Runtime/Injection/ClassInjector.cs

```diff
@@ -128,6 +128,13 @@ public static bool IsTypeRegisteredInIl2Cpp(Type type)
         var currentPointer = Il2CppClassPointerStore.GetNativeClassPointer(type);
         if (currentPointer != IntPtr.Zero)
             return true;
+        if (IsTypeInjectedOnly(type)) return true;
+
+        return false;
+    }
+
+    internal static bool IsTypeInjectedOnly(Type type)
```

Comment:
```suggestion
    internal static bool IsManagedTypeInjected(Type type)
```

Reply:
We also have `RuntimeSpecificsStore.IsInjected`, you might want to use that instead.

Reply:
Then I would need to get the native class ptr. This feels more convinient.